app/services/solr_mapper.rb

```diff
@@ -88,6 +104,39 @@ def funders_ids_field
     metadata['funding_references']&.pluck('funder_identifier')&.compact || []
   end
 
+  def descriptions_by_type_field(description_types)
+    metadata['descriptions']&.filter_map do |d|
+      d['description'].truncate(32_000) if description_types.include?(d['description_type'])
+    end || []
+  end
+
+  def rights_uris_field
+    metadata['rights_list'].pluck('rights_uri')&.uniq&.compact if metadata['rights_list'].present?
```

Comment:
Can you make the `if ...` a guard statement? It is easier to comprehend.

Reply:
Yes. done